dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java

```diff
@@ -157,6 +157,7 @@ public Device findDeviceByTenantIdAndName(TenantId tenantId, String name) {
                 () -> deviceDao.findDeviceByTenantIdAndName(tenantId.getId(), name).orElse(null), true);
     }
 
+    @Transactional
```

Comment:
I think we should mark with `@Transactional` all the methods that use `doSaveDevice`.